src/content/components/TimelineViewport.css

```diff
@@ -0,0 +1,73 @@
+.timelineViewport {
+  flex: 1;
+  position: relative;
+  margin-top: 0;
+  transition: transform 100ms;
+  overflow: hidden;
+}
+
+.timelineViewport.expanded {
+  cursor: -webkit-grab;
+  cursor: grab;
+  border-top: 1px solid #d6d6d6;
+  border-bottom: 1px solid #d6d6d6;
+  margin-bottom: 5px;
+}
+
+.timelineViewport.expanded::after {
+  content: "";
+  pointer-events: none;
+  position: absolute;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  /* Slight blue shadow. */
+  box-shadow: inset 0 0 20px rgba(64, 115, 140, 0.2);
+  z-index:1;
+}
+
+.timelineViewport.collapsed {
+  transform: translateY(9px);
+}
+
+.timelineViewport.dragging {
+  cursor: -webkit-grabbing;
+  cursor: grabbing;
+}
+
+.timelineViewportShiftScroll {
+  position: absolute;
+  pointer-events: none;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  will-change: opacity, transform;
+  bottom: 0;
+  right: 0;
+  padding: 3px 10px;
+  border-radius: 5px 0 0 0;
+  opacity: 1;
+  background: rgba(0,0,0,0.07);
+  line-height: 20px;
+  transition: transform .2s ease-in-out, opacity .2s ease-in-out;
+  box-shadow: 0 0 0 0.5px rgba(0,0,0,0.1), 0 2px 4px rgba(0,0,0,0.1);
+}
+
+.timelineViewportShiftScroll.hidden {
+  opacity: 0;
+  transform: translateY(30px);
+}
+
+.timelineViewportShiftScroll kbd {
```

Comment:
Please give these kbd elements a className so that you don't need the descendant selector here.